### PR TITLE
chore: Build the Python services with Bazel for the Dockerized AGW

### LINKED ISSUE #15073: Use bazel-built services in `lte/gateway/docker/services/python/Dockerfile`
Use bazel-built (Python) services in lte/gateway/docker/services/python/Dockerfile. This should make the Make Python environment/protos/swagger build unnecessary.

### PR DESCRIPTION
**Pull request by LKreutzer**
## Summary

- Remove the buildenv steps from Python Docker build.
- Build Python services/scripts with Bazel.
  - The `tar.gz` is needed to transfer the services with all dependencies (pip etc.) to the runner image.
  - This also uses the same logic as the `.deb` build in production.
  - See also https://github.com/magma/magma/issues/15034#issuecomment-1441909179
- The steps in `_start_gateway_containerized` (fabric) are adapted to what is already documented for local execution in the `lte/gateway/docker/README.md` file. 
- Resolves https://github.com/magma/magma/issues/15073
  - Part of https://github.com/magma/magma/issues/15034
  

## Test Plan

- [Containerized integ test run with dhcp 1](https://github.com/magma/magma/actions/runs/4292251268)
- [Containerized integ test run with dhcp 2](https://github.com/magma/magma/actions/runs/4292252451) 
- [Containerized integ test run with dhcp 3](https://github.com/magma/magma/actions/runs/4293144107) 
- [Containerized integ test run with dhcp 4](https://github.com/magma/magma/actions/runs/4293145134) 

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -55,9 +56,8 @@
 !protos/
 
 !bazel/
-!*.bzl
-!*.bazel
-!*.BUILD
+!**/*.bzl
+!**/*.bazel
```
Comment (LKreutzer) [inline .dockerignore:60]: In the current file this is only valid for the top-level files. `.BUILD` files exist only in the `bazel` folder.

File: .dockerignore [hunk 2]
```diff
@@ -21,6 +21,7 @@
 !lte/gateway/docker/
 !lte/gateway/deploy/roles/magma/files/
 !lte/gateway/python/
+!lte/gateway/release/
```
Comment (LKreutzer) [inline .dockerignore:24]: Needed for the Docker build.

File: lte/gateway/fabfile.py [hunk 3]
```diff
@@ -454,15 +454,16 @@ def integ_test_containerized(
...
-    with c_gw.cd(AGW_PYTHON_ROOT):
-        c_gw.run('make buildenv')
 
-    with c_gw.cd(AGW_ROOT):
-        c_gw.run(
-            'for component in redis nghttpx td-agent-bit; do cp "${MAGMA_ROOT}"'
-            '/{orc8r,lte}/gateway/configs/templates/${component}.conf.template;'
-            ' done',
-        )
+    c_gw.run('sudo rm -rf /etc/snowflake && sudo touch /etc/snowflake')
```
Comment (LKreutzer) [inline lte/gateway/fabfile.py:458]: See `lte/gateway/docker/README.md`

File: lte/gateway/fabfile.py [hunk 4]
```diff
@@ -454,15 +454,16 @@ def integ_test_containerized(
...
+    c_gw.run('cd ${MAGMA_ROOT} && bazel/scripts/link_scripts_for_bazel_integ_tests.sh')
+    c_gw.run('cd ${MAGMA_ROOT} && bazel build `bazel query "attr(tags, util_script, kind(.*_binary,//orc8r/... union //lte/...))"`')
```
Comment (wolfseb) [inline lte/gateway/fabfile.py]: Nit: you could put these two lines in a `with c_gw.cd("${MAGMA_ROOT}")` context instead of manually using `cd` in each run command.

**Comment by MoritzThomasHuebner** [comment]
It looks like the `test_attach_detach_with_non_nat_dhcp`, `test_attach_detach_with_non_nat_dhcp_multi_ue`, `test_attach_detach_with_non_nat_dhcp_multi_looped` are consistently failing in the test runs that you posted. I am not sure what is going wrong at this point, but we should investigate this.

File: lte/gateway/docker/services/python/Dockerfile [hunk 5]
```diff
@@ -18,68 +18,46 @@ ARG CPU_ARCH=x86_64
...
+RUN  wget -P /usr/sbin --progress=dot:giga https://github.com/bazelbuild/bazelisk/releases/download/v1.10.0/bazelisk-linux-"${DEB_PORT}" && \
+  chmod +x /usr/sbin/bazelisk-linux-"${DEB_PORT}" && \
+  ln -s /usr/sbin/bazelisk-linux-"${DEB_PORT}" /usr/sbin/bazel
 
-COPY \
-    ./lte/gateway/python/Makefile \
-    ./lte/gateway/python/defs.mk \
-    ./lte/gateway/python/setup.py \
-    $MAGMA_ROOT/lte/gateway/python/
+COPY ./lte/gateway/python/integ_tests/s1aptests/ovs $MAGMA_ROOT/lte/gateway/python/integ_tests/s1aptests/ovs
```
Comment (MoritzThomasHuebner) [inline lte/gateway/docker/services/python/Dockerfile:51]: I'm curious as to why we specifically are moving the `ovs` directory. Why is this necessary now when it wasn't before. And why aren't we just moving the entire `integ_test` directory just as we do with the `load_tests` directory.

Comment (LKreutzer) [reply lte/gateway/docker/services/python/Dockerfile:51]: It is used by the `packet_ryu_cli` target, which is a script that uses the `rest_api` in the `ovs` folder.

I don't know exactly why the load tests are copied here, I think the argument is that they might be useful in production deployments.

**Comment by LKreutzer** [comment]
> It looks like the `test_attach_detach_with_non_nat_dhcp`, `test_attach_detach_with_non_nat_dhcp_multi_ue`, `test_attach_detach_with_non_nat_dhcp_multi_looped` are consistently failing in the test runs that you posted. I am not sure what is going wrong at this point, but we should investigate this.

I changed it to also build the dhcp helper cli with Bazel, the dhcp tests are green now, see runs in the description.

File: lte/gateway/fabfile.py (hunk 4 above)
Comment (LKreutzer) [reply lte/gateway/fabfile.py]: Done

**Review by MoritzThomasHuebner — APPROVED** [review]

**Review by mpfirrmann — APPROVED** [review]
LGTM
